pkg/controllers/cassandra/nodepool/nodepool.go

```diff
@@ -46,14 +44,8 @@ func (e *defaultCassandraClusterNodepoolControl) updateStatus(cluster *v1alpha1.
 		return err
 	}
 	// Create a NodePoolStatus for each statefulset that is controlled by this cluster.
-	for ssName, ss := range sets {
-		clusterName, npName, err := util.ParseNodePoolResourceName(ssName)
-		if err != nil {
-			glog.Errorf("Error parsing statefulset name: %s", err)
-		}
-		if clusterName != cluster.Name {
-			glog.Errorf("statefulset name %q did not contain cluster name %q", ssName, cluster.Name)
-		}
+	for _, ss := range sets {
+		npName := ss.Labels[util.NodePoolNameLabelKey]
```

Comment:
What if the label isn't set?

Eventually I think this sort of thing should be moved into some kind of 'meta accessor' helper... Not sure how that's best to be structured. We should dig into the upstream codebase. Doesn't need to be for this PR though.

Reply:
Yeah, I don't know where to draw the line in handling cases where an outside program or person has meddled with the labels.  If there isn't a label there'll be a `CassandraCluster.Status.NodePools[""]` NodePoolStatus entry which should give a clue to what's gone wrong.

And if we do some day change the statefulset name format we may and up with multiple sets with the same nodepool label.

I'll leave it like this for now.